src/screens/MainScreen/index.js

```diff
@@ -38,34 +46,36 @@ const MainScreen = ({ navigation }) => {
 
   return (
     <SafeAreaView style={styles.container} testID={MAIN_SCREEN}>
-      <MapView
-        ref={mapView}
-        onLayout={() => setIsMapRendered(true)}
-        provider={PROVIDER_GOOGLE}
-        style={styles.map}
-        initialRegion={{
-          ...location,
-          ...DELTA_COORDS,
-        }}
-        showUserLocation
-        followUserLocation>
-        {topics &&
-          targets.map(({ target: { id, lat, lng, topicId } }) => (
-            <TargetMarker
-              key={id}
-              id={id.toString()}
-              lat={lat}
-              lng={lng}
-              topicId={topicId}
-              onPress={() => onSelectTarget(id, targets)}
-              selected={selectedTarget?.id === id}
-            />
-          ))}
-        <Marker coordinate={location} style={styles.marker}>
-          <Image source={locationMap} style={styles.markerImage} />
-          <View style={styles.markerCircle} />
-        </Marker>
-      </MapView>
+      <View style={styles.mapContainer}>
+        <MapView
+          ref={mapView}
+          onLayout={() => setIsMapRendered(true)}
+          onMapReady={() => mapView.current.animateToRegion(location)}
+          provider={PROVIDER_GOOGLE}
+          style={styles.map}
+          onRegionChangeComplete={region => !formVisible && setLocation(region)}
+          loadingEnabled
+          loadingIndicatorColor={BLACK}>
+          {topics &&
+            targets.map(({ target: { id, lat, lng, topicId } }) => (
+              <TargetMarker
+                key={id}
+                id={id.toString()}
+                lat={lat}
+                lng={lng}
```

Comment:
I see you are passing all the target props you destructured, you can pass the target object directly and then destructure it inside `TargetMarket`